docs/GoogleSheets-batchsource.md

```diff
@@ -0,0 +1,113 @@
+
+**Client ID:** OAuth2 client id.
+
+**Client Secret:** OAuth2 client secret.
+
+**Refresh Token:** OAuth2 refresh token.
+
+**Access Token:** OAuth2 access token.
+
+#### Service Account Properties
+
+**Account File Path:** Path on the local file system of the service account key used for authorization. 
+Can be set to 'auto-detect' for getting service account from system variable.
+The file/system variable must be present on every node in the cluster.
+Service account json can be generated on Google Cloud 
+[Service Account page](https://console.cloud.google.com/iam-admin/serviceaccounts)
+
+### Retrying
+
+**Max Retry Count:** Maximum number of retry attempts.
+
+**Max Retry Wait:** Maximum wait time for attempt in seconds. Initial wait time is one second and it grows exponentially.
+
+**Max Retry Jitter Wait:** Maximum additional wait time in milliseconds.
+
+### Metadata Extraction
+
+**Extract Metadata:** Field to enable metadata extraction. Metadata extraction is useful when user wants to specify 
+a header or a footer for a sheet. The rows in headers and footers are not available as data records. 
+Instead, they are available in every record as a field called 'metadata', which is a record of the specified metadata.
+
+**Metadata Record Name:** Name of the record with metadata content. 
+It is needed to distinguish metadata record from possible column with the same name.
+
+**First Header Row:** Row number of the first row to be treated as header.
+
+**Last Header Row:** Row number of the last row to be treated as header.
+
+**First Footer Row:** Row number of the first row to be treated as footer.
+
+**Last Footer Row:** Row number of the last row to be treated as footer.
+
+**Metadata Cells:** Set of the cells for key-value pairs to extract as metadata from the specified metadata sections.
+Only shown if **Extract metadata** is set to true. The cell numbers should be within the header or footer.
+
+### Advanced
+
+**Text Formatting:** Output format for numeric sheet cells. 
+In 'Formatted values' case the value will contain appropriate format of source cell e.g. '1.23$', '123%'." 
+For 'Values only' only number value will be returned.
+
+**Skip Empty Data:** Field that allows skipping of empty structure records.
+
+**Add spreadSheet/sheet name fields:** Toggle that defines if the sink extends output schema with 
+spreadSheet and sheet names. For _Yes_ selection the spreadSheet and sheet name fields can be specified via 
+**SpreadSheet field name** and **Sheet field name** properties. 
+
+**SpreadSheet field name:** Schema field name for spreadSheet name.
+
+**Sheet field name:** Schema field name for sheet name.
+
+**Column Names Selection:** Source for column names.
+
+**Custom Row For Column Names:** Row number of the row to be treated as a header.
+Only shown when the 'Column Names Selection' field is set to 'Custom row as column names' header.
+
+**Last Data Column:** Last column number of the maximal field of plugin work for data.
```

Comment:
Add "Index" to all field names that are indexes.

Reply:
done